packages/material-ui-lab/src/ToggleButton/ToggleButton.js

```diff
@@ -3,7 +3,8 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { fade, withStyles } from '@material-ui/core/styles';
+import { withStyles } from '@material-ui/core/styles';
+import { fade } from '@material-ui/core/styles/colorManipulator';
```

Comment:
@oliviertassinari  This will pull in our commonJS build. 

RemindMe: Write a lint rule that only allows 1st level path imports

Reply:
It deoptimizes it. Thanks!

Reply:
💯 for the eslint rule.

Reply:
Yeah this adds a lot of confusion. It's not obvious what imports are fine and which are not. 

For v5 we should revisit our packaging strategy so that people don't even have to choose how they import from our package.

Reply:
> For v5 we should revisit our packaging strategy so that people don't even have to choose how they import from our package.

What direction do you have in mind? I wish we were documenting:
```js
import { TextField, makeStyles, Theme, PaperProps } from '@material-ui/core';
```

Reply:
Or `@material-ui/TextField`. Either way I think an option for the import style is helpful. We can revisit this in a separate issue. Before a change I want to give users some time to give feedback in the form of the import style they use. This change should come with a codemod.

Reply:
> Or @material-ui/TextField. 

A past effort: #9532.

Reply:
I have found it. I thought it was more detailed, but it seems it just me arguing against myself: https://github.com/mui-org/material-ui/issues/7839#issuecomment-388138137.